File: 8-DictionariesAndMaps/Solution.py
num = int(input())

phone_book = {}
for i in range(num):
    entry = str(input()).split(" ")

    name = entry[0]
    phone = int(entry[1])
    phone_book[name] = phone

while True:
    try:
        name = input()

        if name in phone_book:
            phone = phone_book[name]
            print(name + "=" + str(phone))
        else:
            print("Not found")
    except:
        break
        

File: 8-DictionariesAndMaps/Solution.cpp
#include <iostream>
#include <map>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    int num;
    string name;    
    
    cin >> num;
    
    map<string, int> phone_book;
    
    for (int i=0; i<num; i++) {

        cin >> name >> phone_book[name];
    }
    
    while (cin >> name){         
        if (phone_book[name]) {
           cout << name << "=" << phone_book[name] << endl;
        } else {
          cout << "Not found" << endl;
        }
    }
    
    return 0;
}